### ISSUE TITLE
Adding external pins feature

### ISSUE THREAD

**Issue by jainpranav1** [davidthings/hdelk#6]
Thank you so much for making this tool. This tool was able to help me make the Nand2Tetris HDL Visualizer extension: https://marketplace.visualstudio.com/items?itemName=PranavJain.nand2tetris-hdl-visualizer&ssr=false#overview which has been deployed in my university's computer organization class.
<img width="960" alt="screenshot1" src="https://user-images.githubusercontent.com/64936978/152648606-398f2a82-c264-4496-8475-d3769aaae8ec.PNG">
In the image, you can see that I modified the HDElk file inorder to accommodate external pins, like sel[0], sel[1], etc. This was done by adding them to internal pins on internal gates, and then shifting the internal pin boxes so that the external pins are separated from the internal pins. Is it possible that you can add a feature to draw external pins within the HDElk library? Then, I wouldn't have to add separate code to do so. Please let me know if this is possible. Thank you again for your time and efforts. This is a really awesome tool.

**Comment by davidthings** [comment]
Could the internal port feature help?

https://davidthings.github.io/hdelk/tutorial#internal-ports

**Comment by jainpranav1** [comment]
Hi! I originally used the internal ports feature, but the ports ended up far from their respective gates connected by long wires. This made the diagram quite cluttered and hard to follow. That is why I decided to not use the internal port feature and instead add my own code to make ports close to their respective gates.
